File: TicTacToe2-java_version/app/src/main/java/com/example/tictactoe/MainActivity.java

package com.example.tictactoe;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

public class MainActivity extends AppCompatActivity {

    public static boolean playerTurn;
    private Button button_Instruction;
    private Button button_Start;
    private Button button_choose;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);



        button_Start = (Button) findViewById(R.id.Start_Game);
        button_Start.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                GameStart();
            }
        });

        button_choose = (Button) findViewById(R.id.Player_choose);
        button_choose.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                PlayerChoose();
            }
        });

        button_Instruction = (Button) findViewById(R.id.game_instruction);
        button_Instruction.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                ShowInstruction();
            }
        });
    }

    public void GameStart(){
        Intent intent2 = new Intent(this,MainActivity2.class);
        startActivity(intent2);
    }

    public void PlayerChoose(){
        Intent intent = new Intent(this, MainActivity3.class);
        startActivity(intent);
    }

    public void ShowInstruction(){
        Intent intent3 = new Intent(this, Main4Activity.class);
        startActivity(intent3);
    }
}

File: DigitClassificationTFLite/app/src/main/java/edu/uw/eep523/digitclassificationtflite/MainActivity.kt
package edu.uw.eep523.digitclassificationtflite

import android.graphics.Color
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import android.util.Log
import android.view.MotionEvent
import android.widget.Button
import android.widget.TextView
import com.divyanshu.draw.widget.DrawView


/// NAME OF YOUR MODEL - tflite file in the asset folder
const val MODEL_FILE = "your_model_name.tflite"


class MainActivity : AppCompatActivity() {

    private var drawView: DrawView? = null
    private var clearButton: Button? = null
    private var predictedTextView: TextView? = null
    private var digitClassifier = DigitClassifier(this)

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        // Setup view instances
        drawView = findViewById(R.id.draw_view)
        drawView?.setStrokeWidth(70.0f)
        drawView?.setColor(Color.WHITE)
        drawView?.setBackgroundColor(Color.BLACK)

        clearButton = findViewById(R.id.clear_button)
        predictedTextView = findViewById(R.id.predicted_text)


        // Setup clear drawing button
        clearButton?.setOnClickListener {
            drawView?.clearCanvas()

            predictedTextView?.text = ""
        }

        // Setup classification trigger so that it classify after every stroke drew
        drawView?.setOnTouchListener { _, event ->
            // As we have interrupted DrawView's touch event,
            // we first need to pass touch events through to the instance for the drawing to show up
            drawView?.onTouchEvent(event)

            // Then if user finished a touch event, run classification
            if (event.action == MotionEvent.ACTION_UP) {
                classifyDrawing()
           }

            true
        }

        // Setup digit classifier
        digitClassifier
                .initialize()
                .addOnFailureListener { e -> Log.e(TAG, "Error to setting up digit classifier.", e) }
    }

    override fun onDestroy() {
        digitClassifier.close()
        super.onDestroy()
    }

    private fun classifyDrawing() {
        val bitmap = drawView?.getBitmap()

        if ((bitmap != null) && (digitClassifier.isInitialized)) {
            digitClassifier
                    .classifyAsync(bitmap) //Call function from DigitClassifier class

                    .addOnSuccessListener { resultText -> predictedTextView?.text = resultText }
                    .addOnFailureListener { e ->
                        predictedTextView?.text = "classification error"
                        Log.e(TAG, "Error classifying drawing.", e)
                    }
        }
    }

    companion object {
        private const val TAG = "MainActivity"
    }
}
